### ISSUE TITLE
Generic property types not inferred

### ISSUE THREAD

**Issue by jackd** [microsoft/pyright#383]
In the following, `FooInt.bar` should be inferred to be an int, but isn't. It works fine if using methods rather than properties.

```python
from typing import TypeVar, Generic

T = TypeVar('T')


class Foo(Generic[T]):

    def __init__(self, bar: T):
        self._bar = bar

    @property
    def bar(self) -> T:
        return self._bar


class FooInt(Foo[int]):
    def fizz(self) -> int:
        return self.bar.bit_length()  # <---
    

    # uncommenting the below silences the error - but should that be necessary?
    # @property
    # def bar(self) -> int:
    #     return super(FooInt, self).bar
```

Error message:
```
bit_length: Unknown
Cannot access member 'bit_length' for type 'TypeVar['T']'
  Unsupported type 'TypeVar['T']'pyright
```

**VS Code extension or command-line**
1.1.7

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the bug, and I think I know what's wrong.

You can work around the issue in the meantime by omitting the "-> T" for the property return type. Pyright correctly infers the return type, but it's not properly handling a generic type argument in this context.

**Comment by jackd** [comment]
Update: it's got nothing to do with subclassing, just properties.

```python
from typing import TypeVar, Generic

T = TypeVar('T')


class Foo(Generic[T]):

    def __init__(self, bar: T):
        self._bar = bar

    @property
    def bar(self) -> T:
        return self._bar

    def bar_method(self) -> T:
        return self._bar


foo = Foo[int](3)
foo.bar.bit_length()  # <--- "Cannot access member 'bit_length' for type 'TypeVar['T']'  Unsupported type 'TypeVar['T']'"
foo.bar_method().bit_length()  # <--- no issues
```

**Comment by erictraut** [comment]
This will be fixed in the next version of pyright. Thanks again for reporting it.

**Comment by jackd** [comment]
Thanks for the quick turn-around :)

**Comment by erictraut** [comment]
This is now fixed in 1.1.8, which I just published.
